drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -830,38 +831,137 @@ local function do_bridge_network_init(driver, device, bridge_url, api_key)
+                  driver:do_hue_light_delete(light_device)
                 end
               end
             end
           elseif event.type == "add" then
             for _, add_data in ipairs(event.data) do
-              if add_data.type == "light" then
+              if add_data.type == "light" and add_data.owner and add_data.owner.rtype == "device" then
                 log.info(
                   string.format(
-                    "New light added to Hue Bridge \"%s\", light properties: \"%s\", " ..
-                    "re-run discovery to join new lights to SmartThings",
-                    device.label, st_utils.stringify_table(add_data, nil, false)
+                    "New light added to Hue Bridge \"%s\", light properties: \"%s\"",
+                    bridge_device.label, json.encode(add_data)
                   )
                 )
+
+                cosock.spawn(function()
+                  local hue_api = bridge_device:get_field(Fields.BRIDGE_API)
+                  if hue_api == nil then
+                    bridge_device.log.warn("No Hue API instance available for new light event.")
+                    return
+                  end
+
+                  local hue_device_rid = add_data.owner.rid
+                  local rest_resp, rest_err = hue_api:get_device_by_id(hue_device_rid)
+
+                  if rest_err ~= nil then
+                    log.error("Error getting device information for new light: " .. tostring(rest_err))
+                    return
+                  end
+
+                  if rest_resp == nil then
+                    log.error("REST Response while handling New Light Event unexpectedly nil without error message")
+                    return
+                  end
+
+                  if rest_resp.errors and #rest_resp.errors > 0 then
+                    for _, hue_error in ipairs(rest_resp.errors) do
+                      log.error_with({ hub_logs = true }, "Error in Hue API response: " .. hue_error.description)
+                    end
+                    return
+                  end
+
+                  local new_device_info = nil
+                  for _, hue_device in ipairs(rest_resp.data or {}) do
+                    for _, svc_info in ipairs(hue_device.services or {}) do
+                      if svc_info.rtype == "light" and svc_info.rid == add_data.id then
+                        new_device_info = hue_device
+                        break
+                      end
+                    end
+                    if new_device_info ~= nil then break end
+                  end
+
+                  if new_device_info == nil then
+                    log.warn("Couldn't get all device info for new light, unable to join. Try using Scan Nearby to find new Hue lights.")
+                    return
+                  end
+
+                  log.info(
+                    string.format(
+                      "Adding light \"%s\" to Philips Hue Edge Driver",
```

Comment:
nit: The driver name will already be logged so probably not necessary to include it in the message